SalesforceFoundation/EDA
# Prevent nested plan requirement having lookup to program plan

# Critical Changes

# Changes
- We've included a custom setting that you can use to ensure that a child Plan Requirement can't look up to a Program Plan. This prevents users from incorrectly including Plan Requirements in a Program Plan that is not the Program Plan used by the parent requirement.
For example, if a Program Plan called B.S. in Biology 2018/2019 has a parent Plan Requirement of Organic Chemistry with child Plan Requirements for Organic Chemistry I and Organic Chemistry II, we wouldn't want either of those child requirements looking up to a Program Plan for B.S. in Forensic Pathology.
This new setting is turned on by default for new customers and is disabled for existing customers. You can access the new setting from under Validate Program Plan for Nested Plan Requirements in EDA Settings | Program Plans | Settings.
# Issues Closed
Fixes #812
# New Metadata
PREQ_PreventPPlanParent_TDTM.cls
PREQ_PreventPPlanParent_TEST
Hierarchy_Settings__c.Validate_Program_Plan_for_Nested_PR__c
STG_CMP_ProgramPlans
# Deleted Metadata

# Testing Notes
Turn on custom setting (Validate Program Plan for Nested Plan Requirements) under EDA setting page -> Program Plan
1. Create a program plan: Test Program Plan 1
2. Create a plan requirement: Test Plan Requirement 1 and populate the program plan field with Test Program Plan 1
3. Go to the related list of Test Plan Requirement 1 and create a child plan requirement: Test Child Plan Requirement 1. Try to fill in the program plan field of the child plan requirement and click save
4. Make sure the record is not saved and the error message is shown
5. Find an existing plan requirement and try to fill in the program plan field. The result should be the same.
